core/src/Cabana_LinkedCellList.hpp

```diff
@@ -75,21 +76,33 @@ class LinkedCellList
       \param grid_min Grid minimum value in each direction.
 
       \param grid_max Grid maximum value in each direction.
+
+      \param neighborhood_radius Radius for neighbors
+
+      \param cell_size_ratio Ratio of the cell size to the neighborhood size
     */
     template <class SliceType>
     LinkedCellList(
         SliceType positions, const typename SliceType::value_type grid_delta[3],
         const typename SliceType::value_type grid_min[3],
         const typename SliceType::value_type grid_max[3],
+        const Scalar neighborhood_radius, const Scalar cell_size_ratio = 1,
         typename std::enable_if<( is_slice<SliceType>::value ), int>::type* =
             0 )
         : _grid( grid_min[0], grid_min[1], grid_min[2], grid_max[0],
                  grid_max[1], grid_max[2], grid_delta[0], grid_delta[1],
                  grid_delta[2] )
+        , _sorted( false )
+        , _rsqr( neighborhood_radius * cell_size_ratio )
     {
         std::size_t np = positions.size();
         allocate( totalBins(), np );
         build( positions, 0, np );
+
+        // Stencil constructor carry-over
```

Comment:
I'm realizing I probably never said it, but I imaged keeping a separate struct and constructing a stencil instance inside the LCL

Reply:
So are you thinking to leave the `linkedCellStencil` mainly as it is and then just storing a copy of it when creating the `linkedCellList`? Then creating a wrapper function similar to those for the grid and bindata functions that calls `getCells` from the stencil stored as a member of the list?

Reply:
Yeah, I think that keeps a good encapsulation